pkg/kubelet/apis/config/v1beta1/defaults.go

```diff
@@ -132,6 +132,9 @@ func SetDefaults_KubeletConfiguration(obj *kubeletconfigv1beta1.KubeletConfigura
 	if obj.NodeLeaseDurationSeconds == 0 {
 		obj.NodeLeaseDurationSeconds = 40
 	}
+	if obj.MaxContainerCheckpoints == nil || *obj.MaxContainerCheckpoints < int32(0) {
```

Comment:
a non-nil value <= 0 should fail validation, not be silently overwritten in defaulting

Reply:
right, will fix

Reply:
fixed